### PR TITLE
fix races

### LINKED ISSUE #8: Multiple race conditions
There are multiple race conditions

```
WARNING: DATA RACE
Write at 0x00c00011e3e0 by goroutine 7:
  github.com/zenthangplus/goccm.(*concurrencyManager).Close()
      /home/tom/src/tomkukral/goccm/goccm.go:109 +0x18d
  github.com/zenthangplus/goccm.(*concurrencyManager).WaitAllDone()
      /home/tom/src/tomkukral/goccm/goccm.go:115 +0x181
  github.com/zenthangplus/goccm.TestExample()
      /home/tom/src/tomkukral/goccm/goccm_test.go:19 +0x180
  testing.tRunner()
      /opt/golang/1.19.3/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /opt/golang/1.19.3/src/testing/testing.go:1493 +0x47

Previous read at 0x00c00011e3e0 by goroutine 8:
  github.com/zenthangplus/goccm.(*concurrencyManager).controller()
      /home/tom/src/tomkukral/goccm/goccm.go:78 +0x88
  github.com/zenthangplus/goccm.New.func1()
      /home/tom/src/tomkukral/goccm/goccm.go:61 +0x39

Goroutine 7 (running) created at:
  testing.(*T).Run()
      /opt/golang/1.19.3/src/testing/testing.go:1493 +0x75d
  testing.runTests.func1()
      /opt/golang/1.19.3/src/testing/testing.go:1846 +0x99
  testing.tRunner()
      /opt/golang/1.19.3/src/testing/testing.go:1446 +0x216
  testing.runTests()
      /opt/golang/1.19.3/src/testing/testing.go:1844 +0x7ec
  testing.(*M).Run()
      /opt/golang/1.19.3/src/testing/testing.go:1726 +0xa84
  main.main()
      _testmain.go:51 +0x2e9

Goroutine 8 (running) created at:
  github.com/zenthangplus/goccm.New()
      /home/tom/src/tomkukral/goccm/goccm.go:61 +0x1d6
  github.com/zenthangplus/goccm.TestExample()
      /home/tom/src/tomkukral/goccm/goccm_test.go:10 +0x2e
  testing.tRunner()
      /opt/golang/1.19.3/src/testing/testing.go:1446 +0x216
  testing.(*T).Run.func1()
      /opt/golang/1.19.3/src/testing/testing.go:1493 +0x47
 ```

### PR DESCRIPTION
**Pull request by tomkukral**
fix test races

closes https://github.com/zenthangplus/goccm/issues/8

### CODE REVIEW

**Comment by tomkukral** [comment]
@zenthangplus please review

File: goccm.go [hunk 1]
```diff
@@ -105,8 +106,24 @@ func (c *concurrencyManager) Done() {
...
+	// terminate if channel is already closed
+	select {
+	case <-c.closed:
+		return
+	default:
+		close(c.closed)
+	}
+}
+
+func (c *concurrencyManager) IsClosed() bool {
```
Comment (zenthangplus) [inline goccm.go:120]: Should we expose this method?

Comment (tomkukral) [reply goccm.go]: makes sense, changing it

**Comment by tomkukral** [comment]
@zenthangplus updated according to your comments, please review again

**Comment by zenthangplus** [comment]
> @zenthangplus updated according to your comments, please review again

@tomkukral test failed, please check

**Comment by tomkukral** [comment]
It is failing on golang 1.10 (released in 2018) and this legacy version doesn't support formating I was using.

There are two options:
* A - stop testing on golang 1.10
* B - update code to be compatible with 1.10

I'd prefer A because 1.10 is very old and many projects have already migrated to newer version. My my suggestion is remove 1.10 from tests and add 1.19. Do you agrees @zenthangplus ?

**Comment by zenthangplus** [comment]
> It is failing on golang 1.10 (released in 2018) and this legacy version doesn't support formating I was using.
> 
> There are two options:
> 
> * A - stop testing on golang 1.10
> * B - update code to be compatible with 1.10
> 
> I'd prefer A because 1.10 is very old and many projects have already migrated to newer version. My my suggestion is remove 1.10 from tests and add 1.19. Do you agrees @zenthangplus ?

@tomkukral i think it not related to v1.10, because you are printed `chan int32` as integer type. could you help to fix it? 
https://github.com/zenthangplus/goccm/pull/9/files#diff-37ca082752c510e847f4d76524b6c582371b834c2007b6bdf17316454eea9c1aR69

**Comment by tomkukral** [comment]
@zenthangplus you are right, it can change type before printing it. Fixed.

**Comment by zenthangplus** [comment]
Thank @tomkukral, it was merged

**Comment by tomkukral** [comment]
Btw, I have implemented very similar logic in [my tool](https://gitlab.com/6shore.net/wq/-/blob/master/ticket/ticket.go). I wanted to get possiblity to limit waiting time and abort waiting. You can check it and use it in this library.

**Pull request merged**